kpatch-build/create-diff-object.c

```diff
@@ -1377,6 +1398,19 @@ void kpatch_verify_patchability(struct kpatch_elf *kelf)
 
 	if (errs)
 		DIFF_FATAL("%d unsupported section change(s)", errs);
+
+	list_for_each_entry(sym, &kelf->symbols, list) {
+		if (sym->type != STT_FUNC || sym->status != CHANGED)
+			continue;
+		if (!sym->has_fentry_call) {
+			log_normal("function %s has no fentry call, unable to patch\n",
+				   sym->name);
+			errs++;
+		}
+	}
```

Comment:
I think this won't work, because kpatch_verify_patchability() is called with the patched object, not the original (base) object.  The base object has the original function which will be patched.

Instead of having this check here, I think a similar check will need to be made after kpatch_compare_correlated_elements() and before kpatch_elf_teardown().  For any patched function symbols with status CHANGED, their twin should have sym->has_fentry_call set.